great_expectations/checkpoint/checkpoint.py

```diff
@@ -239,6 +252,14 @@ def save(self) -> None:
 
         store.update(key=key, value=self)
 
+    def _save_before_run(self) -> None:
```

Comment:
I think @billdirks mentioned this in the first pass at this work, but why is this a separate method from `save`? I think the best move would be to remove this, but then comment the "why" when we call save.

Edit: I see - save updates, this adds. The name still feels off though. very non-blocking: the functionality just happens to be used before running the checkpoint, but isn't really about that. Maybe `_save_to_store` and a docstring on why we need to do this?

Reply:
Since @cdkini is out I'll make this change so we can merge. Maybe I'll name to `_add_to_store` since `save` does and update.